File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/PregReplace.java
package com.onevgo.functions;

import java.util.regex.Pattern;

public class PregReplace {
    public static String pregReplace(String pattern, String replacement, String subject) {
        return pregReplace(Pattern.compile(pattern), replacement, subject);
    }

    public static String pregReplace(Pattern pattern, String replacement, String subject) {
        return pattern.matcher(subject).replaceAll(replacement);
    }

    public static void main(String[] args) {
        String string = "April 15, 2003";
        String pattern = "(\\w+) (\\d+), (\\d+)";
        String replacement = "$1\\1,$3";
        System.out.println(pregReplace(pattern, replacement, string));
    }
}

File: 01_Language/01_Functions/python/preg_replace.py
# coding: utf-8

import re


def preg_replace(pattern, replacement, subject):
    return re.sub(pattern, replacement, subject)


if __name__ == '__main__':
    print(preg_replace(r'(\w+) (\d+), (\d+)', r'\g<1>1,\g<3>', 'April 15, 2003'))
